climakitae/core.py

```diff
@@ -132,19 +132,39 @@ def load(self, data):
         return _compute(data)
 
     # === Retrieve ===================================
-    def retrieve(self):
+    def retrieve(self, config=None, merge=True):
         """Retrieve data from catalog
-
-        Applications.selections and Applications.location determine data retrieves
-        Grabs the data from the AWS S3 bucket, returns lazily loaded dask array
-        User-facing function that provides a wrapper for _read_from_catalog
-
+        
+        By default, Application.selections and Application.location determines the data retrieved. 
+        To retrieve data using the settings in a configuration csv file, set config to the local 
+        filepath of the csv.
+        Grabs the data from the AWS S3 bucket, returns lazily loaded dask array.
+        User-facing function that provides a wrapper for _read_from_catalog and _read_data_from_csv. 
+        
+        Parameters
+        ----------
+        config: str, optional. 
+            Local filepath to configuration csv file
+            Default to None-- retrieve settings in app.selections and app.location
+        merge: bool, optional
+            If config is TRUE and multiple datasets desired, merge to form a single object?
+            Defaults to True.
+        
         Returns
         -------
         xr.DataArray
             Lazily loaded dask array
 
         """
+        if config is not None:
+            if type(config) == str:
+                return _read_data_from_csv(
+                    self.selections, self.location, self._cat, config, merge
+                )
+            else:
+                raise ValueError(
+                    "To retrieve data from a configuration file, please input the path to your local configuration csv as a string"
+                )
         return _read_from_catalog(self.selections, self.location, self._cat)
 
     def retrieve_from_csv(self, csv, merge=True):
```

Comment:
can make this _retrieve_from_csv since the user will no longer ever call it directly

Reply:
I left it in core since it's used as app.retrieve_from_csv() in several of our existing DFU notebooks. Will need to replace those calls with app.retrieve(config="blahblah.cvs") in a PR to cae-notebooks, but want to merge the DFU notebooks from Thur to main first.